frontend/amundsen_application/static/js/components/Alert/AlertList.tsx

```diff
@@ -3,28 +3,67 @@
 
 import * as React from 'react';
 
-import { NoticeType } from 'config/config-types';
+import { NoticeSeverity, NoticeType } from 'config/config-types';
 import { Alert } from './Alert';
 
 export interface AlertListProps {
   notices: NoticeType[];
 }
 
+export interface AggregatedAlertListProps {
+  notices: {
+    [key: string]: NoticeType;
+  };
+}
+
+const aggregateNotices = (notices) =>
+  notices.reduce((accum, notice: NoticeType) => {
+    if (notice) {
+      const { messageHtml, severity, payload } = notice;
+
+      if (typeof messageHtml !== 'function') {
```

Comment:
Only case I'm not sure of is when `messageHtml` is a function and takes a `resourceName`. Not really sure what I would use for that case but if anyone has any ideas I can do it or if we assume it's not happening for this we can just ignore it I guess.

Reply:
good question, I just checked and we have one instance where we were using this functionality at Lyft. I looked into this a bit and I don't think it would ever be a function type at this point. I included my investigation below, but TLDR I think this is fine to leave as you have it here

in the table detail page it passes the results from this function [aggregateResourceNotices](https://github.com/amundsen-io/amundsen/blob/890bfaf92c4b01ff524791b0447f2f57fee1b0a1/frontend/amundsen_application/static/js/pages/TableDetailPage/index.tsx#L122) to the AlertList, and that function calls [getResourceNotices](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-utils.ts#L114). in there it transforms the notices that have function types as their messageHtml to call the function and set set the field to its string message. I investigated backward from the original implementation of function messageHtml fields [here](https://github.com/amundsen-io/amundsen/pull/1280/files)